policies/policies.go

```diff
@@ -0,0 +1,301 @@
+	}
+
+	identityToken, err := metadata.Get(config.IdentityTokenPath)
+	if err != nil {
+		return nil, err
+	}
+
+	req := &osconfigpb.LookupEffectiveGuestPoliciesRequest{
+		Instance:        instance,
+		OsShortName:     shortName,
+		OsVersion:       version,
+		OsArchitecture:  arch,
+		InstanceIdToken: identityToken,
+	}
+	logger.Debugf("LookupEffectiveGuestPolicies request: {Instance: %s, OsShortName: %s, OsVersion: %s, OsArchitecture: %s}",
+		req.GetInstance(), req.GetOsShortName(), req.GetOsVersion(), req.GetOsArchitecture())
+
+	res, err := client.LookupEffectiveGuestPolicies(ctx, req)
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			return nil, fmt.Errorf("code: %q, message: %q, details: %q", s.Code(), s.Message(), s.Details())
+		}
+		return nil, err
+	}
+	logger.Debugf("LookupEffectiveGuestPolicies response:\n%s\n\n", dump.Sprint(res))
+
+	return res, nil
+}
+
+func setConfig(res *osconfigpb.LookupEffectiveGuestPoliciesResponse) error {
+	var aptRepos []*osconfigpb.AptRepository
+	var yumRepos []*osconfigpb.YumRepository
+	var zypperRepos []*osconfigpb.ZypperRepository
+	var gooRepos []*osconfigpb.GooRepository
+	for _, repo := range res.GetPackageRepositories() {
+		if r := repo.GetPackageRepository().GetGoo(); r != nil {
+			gooRepos = append(gooRepos, r)
+			continue
+		}
+		if r := repo.GetPackageRepository().GetApt(); r != nil {
+			aptRepos = append(aptRepos, r)
+			continue
+		}
+		if r := repo.GetPackageRepository().GetYum(); r != nil {
+			yumRepos = append(yumRepos, r)
+			continue
+		}
+		if r := repo.GetPackageRepository().GetZypper(); r != nil {
+			zypperRepos = append(zypperRepos, r)
+			continue
+		}
+	}
+
+	var aptInstalled, aptRemoved, aptUpdated []*osconfigpb.Package
+	var yumInstalled, yumRemoved, yumUpdated []*osconfigpb.Package
+	var zypperInstalled, zypperRemoved, zypperUpdated []*osconfigpb.Package
+	var gooInstalled, gooRemoved, gooUpdated []*osconfigpb.Package
+	for _, pkg := range res.GetPackages() {
+		switch pkg.GetPackage().GetManager() {
+		case osconfigpb.Package_ANY:
+			switch pkg.GetPackage().GetDesiredState() {
+			case osconfigpb.DesiredState_INSTALLED:
+				gooInstalled = append(gooInstalled, pkg.GetPackage())
+				aptInstalled = append(aptInstalled, pkg.GetPackage())
+				yumInstalled = append(yumInstalled, pkg.GetPackage())
+				zypperInstalled = append(zypperInstalled, pkg.GetPackage())
+			case osconfigpb.DesiredState_REMOVED:
+				gooRemoved = append(gooRemoved, pkg.GetPackage())
+				aptRemoved = append(aptRemoved, pkg.GetPackage())
+				yumRemoved = append(yumRemoved, pkg.GetPackage())
+				zypperRemoved = append(zypperRemoved, pkg.GetPackage())
+			case osconfigpb.DesiredState_UPDATED:
+				gooUpdated = append(gooUpdated, pkg.GetPackage())
+				aptUpdated = append(aptUpdated, pkg.GetPackage())
+				yumUpdated = append(yumUpdated, pkg.GetPackage())
+				zypperUpdated = append(zypperUpdated, pkg.GetPackage())
```

Comment:
consider DRYing this up with a function like `func addPackage(goo, apt, yum, zypper) (goo, apt, yum, zypper)`